js-miniapp-bridge/src/common-bridge.ts

```diff
@@ -0,0 +1,120 @@
+/* tslint:disable:no-any */
+const mabMessageQueue: Callback[] = [];
+export { mabMessageQueue };
+
+export interface Callback {
+  id: string;
+
+  onSuccess: (value: string) => void;
+  onError: (error: string) => void;
+}
+
+export class MiniAppBridge {}
+
+export interface MiniAppBridge {
+  /**
+   * Method to call the native interface methods for respective platforms
+   * such as iOS & Android
+   * @param  {[String]} action Action command/interface name that native side need to execute
+   * @param  {Object} param Object that contains request parameter values like permissions.
+   * For eg., {permission: 'location'}
+   * @param  {[Function]} onSuccess Success callback function
+   * @param  {[Function]} onError Error callback function
+   */
+  exec(
+    action: string,
+    param: any,
+    onSuccess: (value: string) => void,
+    onError: (error: string) => void
+  ): void;
+
+  /**
+   * Success Callback method that will be called from native side
+   * to this bridge. This method will send back the value to the
+   * mini apps that uses promises
+   * @param  {[String]} messageId Message ID which will be used to get callback object from messageQueue
+   * @param  {[String]} value Response value sent from the native on invoking the action command
+   */
+  execSuccessCallback(messageId: string, value: string): void;
+
+  /**
+   * Error Callback method that will be called from native side
+   * to this bridge. This method will send back the error message to the
+   * mini apps that uses promises
+   * @param  {[String]} messageId Message ID which will be used to get callback object from messageQueue
+   * @param  {[String]} errorMessage Error message sent from the native on invoking the action command
+   */
+  execErrorCallback(messageId: string, errorMessage: string): void;
+
+  /**
+   * Associating getUniqueId function to MiniAppBridge object
+   */
+  getUniqueId(): Promise<string>;
+
+  /**
+   * Associating requestPermission function to MiniAppBridge object
+   * @param {String} permissionType Type of permission that is requested. For eg., location
+   */
+  requestPermission(permissionType: string): void;
+}
+
+MiniAppBridge.prototype.execSuccessCallback = (messageId, value) => {
```

Comment:
Adding to the `prototype` works, but it seems we lose some of the benefits of Typescript. See my above comment about defining these as class methods and then instantiating `new MiniAppBridge()`